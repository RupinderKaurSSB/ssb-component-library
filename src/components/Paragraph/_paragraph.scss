.ssb-paragraph {
  @include open-sans;
  color: $ssb-dark-6;
  font-size: 16px;
  line-height: 1.7;
  max-width: $max-readable-width;

  &.negative {
    color: $ssb-white;
  }
}
