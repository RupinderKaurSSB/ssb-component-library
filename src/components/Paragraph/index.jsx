import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { openSans } from '../../style/mixins';
import { ssbDark6 } from '../../style/colors';

const Paragraph = ({ children }) => <Wrapper className="paragraph-wrapper">{ children }</Wrapper>;

const Wrapper = styled.p`
	${openSans};
	color: ${ssbDark6};
	font-size: 16px;
	line-height: 1.7;
`;

Paragraph.propTypes = { children: PropTypes.node };

export default Paragraph;
