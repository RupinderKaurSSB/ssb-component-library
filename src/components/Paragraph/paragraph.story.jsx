import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';
import Paragraph from './index';

storiesOf('Paragraph', module).addDecorator(centered)
	.add('Standard', () => (
		<Paragraph>
			A paragraph is useful for wrapping text.
		</Paragraph>
	));
