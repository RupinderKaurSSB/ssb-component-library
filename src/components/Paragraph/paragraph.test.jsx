import React from 'react';
import {shallow} from 'enzyme';
import Paragraph from './index';

describe('Paragraph component', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<Paragraph>Paragraph</Paragraph>);
		expect(wrapper).toMatchSnapshot();
	});
	test('Toggles classNames correctly', () => {
		const wrapper = shallow(<Paragraph className="mt-3" negative>Paragraph</Paragraph>);
		expect(wrapper.hasClass('mt-3')).toEqual(true);
		expect(wrapper.hasClass('negative')).toEqual(true);
	});

});
