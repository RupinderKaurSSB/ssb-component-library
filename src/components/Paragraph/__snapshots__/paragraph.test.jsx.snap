// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph component Matches the snapshot 1`] = `
<p
  className="ssb-paragraph"
>
  Paragraph
</p>
`;
